docs/WindowsSubsystemForLinux.md

```diff
@@ -0,0 +1,94 @@
+
+```
+info: =======================================
+info: Listening on http://localhost:10240/
+info:   serving static files from 'static'
+info:   git release bbf1407109d0439199f71bfdf4037fdeb0eb8393
+info: =======================================
+```
+
+Now you can point your favorite web browser at http://localhost:10240 and see your own personal CE in action!
+
+## Code changes
+
+CE only required a few changes in order to run properly under WSL. Those changes are listed here:
+
+- `app.js`: 
+  - `process.env.wsl` is set if CE if the string "Microsoft" in found in the output of `uname -a`. This works for all WSL distros as they all run on the base Microsoft Linux kernel. 
+  - If the `-tmpDir` option is specified on the command line, both `process.env.tmpDir` and `process.env.winTmp` are set to the specified value Note that if this is specified as a non-Windows volume, Windows executables will fail to run properly. Otherwise, `process.env.winTmp` is set to the value of the Windows `%TEMP%` directory.
+- `lib/base-compiler.js`: Create temporary directories in both `tmpDir` (or the default Linux temporary directory if unspecified) and `winTmp`. CEs creates directories under the temp directory that start with `compiler-explorer-compiler` where the compiland and compiler output lives. 
+- `lib/exec.js`: Execute the compiler in the temporary directory. If the compiler's binary is located on a mounted volume (`startsWith("/mnt"`)) and CE is running under WSL, run the compiler in the `winTmp` directory. Otherwise, use the Linux temp directory. 
+- `lib/compilers/WSL-CL.js`: See also `Wine-CL.js`, the Wine version of this compiler-specific file. This one is used for MSVC on Windows. Its main operation is to translate from Linux-style directories to Windows-style directories (`/mnt/c/tmp` to `c:/tmp`) so that `CL.exe` can find its input files. 
+- `etc/config/c++.defaults.properties`: Add a configuration (`&cl19`) for MSVC compilers. This edits in here are currently wrong in two ways, but it doesn't affect the main CE instance as it uses `amazon` proprties files, and it doesn't affect anyone running a local copy of CE because CE will just fail silently when it can't find a compiler.
+  - The locations of these are hardcoded to a particular install location. See **MSVC setup** below for more information.
+  - Setting of the `%INCLUDE%` path is done with the `/I` switch. This is very clunky and will fall over when command-line limits are hit but it's the only option currently as environments aren't passed through when starting a Windows process from WSL. 
+
+## Debugging
+
+The only viable option for debugging under WSL is to use [VS Code](https://code.visualstudio.com). Because VS Code doesn't currently run natively under WSL, you have to attach to a running CE instance. The following is a `launch.json` that works for attaching to an instance of CE that was launched with the `--inspect` flag. 
+
+```json
```

Comment:
Note that this is not json (No comments in json), and GitHub is complaining. So you either move the comments as a note after the code snippet, or remove the json declaration (and lose the colouring)

Reply:
I can see that now. I don't think the comments are needed at all (nor is the colorization). I'll fix it to make GitHub happy.